src/Form/Expression/ContextFormTrait.php

```diff
@@ -49,7 +49,7 @@ public function buildContextForm(array $form, FormStateInterface $form_state, $c
       $default_value = $context_definition->getDefaultValue();
     }
     $form['context'][$context_name]['setting'] = [
-      '#type' => isset($context_definition->formElement) ? $context_definition->formElement : 'textfield',
+      '#type' => $context_definition->getFormElement(),
       '#title' => $title,
       '#required' => $context_definition->isRequired(),
```

Comment:
formElement is a protected property, so this wasn't working. I've added a 'public' getter method so this should now work. Also the default formElement is textfield already so no need for an 'isset'.